fission/src/systems/match_mode/RobotDimensionTracker.ts

```diff
@@ -0,0 +1,41 @@
+import MirabufSceneObject from "@/mirabuf/MirabufSceneObject"
+import { MiraType } from "@/mirabuf/MirabufLoader"
+import SceneRenderer from "../scene/SceneRenderer"
+import MatchMode from "./MatchMode"
+import SimulationSystem from "@/systems/simulation/SimulationSystem"
+
+const PENALTY_COOLDOWN = 1000
+
+class RobotDimensionTracker {
+    private static _robotHeightPenalties: Map<number, number> = new Map()
+    private static _ignoreRotation: boolean = true
+    private static _maxHeight: number = Infinity
+    private static _heightPenalty: number = 0
+
+    public static setConfigValues(ignoreRotation: boolean, maxHeight: number, heightPenalty: number) {
+        this._ignoreRotation = ignoreRotation
+        this._maxHeight = maxHeight
+        this._heightPenalty = heightPenalty
+    }
+
+    public static update(sceneRenderer: SceneRenderer): void {
+        if (!MatchMode.getInstance().isMatchEnabled()) return
+
+        const robots = [...sceneRenderer.sceneObjects.values()].filter(
+            (obj): obj is MirabufSceneObject => obj instanceof MirabufSceneObject && obj.miraType === MiraType.ROBOT
+        )
+
+        robots.forEach(robot => {
+            const dimensions = this._ignoreRotation ? robot.getDimensionsWithoutRotation() : robot.getDimensions()
+
+            if (dimensions.height > this._maxHeight) {
```

Comment:
Consider adding a small epsilon buffer here. I understand that this much precision isn't really required for our application but its best practice.